massa-pos-exports/src/controller_traits.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+//! This module exports generic traits representing interfaces for interacting
+//! with the PoS selector worker.
+
+use crate::CycleInfo;
+use crate::Selection;
+use anyhow::Result;
+use massa_models::Address;
+use massa_models::Slot;
+
+/// interface that communicates with the selector worker thread
+pub trait SelectorController: Send + Sync {
+    /// Feed cycle to the selector
+    ///
+    /// # Arguments
+    /// * `cycle_info`: give or regive a cycle info for a background
+    ///                 computation of the draws.
+    fn feed_cycle(&self, cycle_info: CycleInfo);
+
+    /// Get [Selection] computed for a slot:
+    /// # Arguments
+    /// * `slot`: target slot of the selection
+    fn get_selection(&self, slot: Slot) -> Result<Selection>;
+
+    /// Get [Address] of the selected block producer for a given slot
+    /// # Arguments
+    /// * `slot`: target slot of the selection
+    fn get_producer(&self, slot: Slot) -> Result<Address>;
+
+    /// Returns a boxed clone of self.
+    /// Useful to allow cloning `Box<dyn SelectorController>`.
+    fn clone_box(&self) -> Box<dyn SelectorController>;
```

Comment:
It could be cleaner to hide this method from the trait(since it is more about the "infra" behind the trait).

You can do that with another trait called for example `SelectorController` like:

```
pub trait SelectorControllerClone {
    fn clone_box(&self) -> Box<dyn SelectorController>;
}
```

and make that a contraint on `SelectorController` itself. 

Then the actual implementing struct can implement `clone_box`. 

The only difference is you remove `clone_box` from `SelectorController`, so that the trait is only about "how to use the controller", as oppose to the basic stuff like "how to clone it".